### PR TITLE
FEI-4916: Add useIsMounted() hook to wonder-blocks-core

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
Functional components sometimes need to know if the component is still mounted.  This can come up with callbacks for asynchronous actions.  Having a hook that encapsulates checking if we're still mounted is a nice quality-of-life improvement when working with functional components and hooks.

Issue: FEI-4916

TODO:
- [x] documentation

## Test plan:
- yarn jest use-is-mounted

**Screenshot of storybook docs:**
Screen Shot 2023-01-12 at 11 33 19 AM

### CODE REVIEW

**Review by jandrade — APPROVED** [review]
LGTM! 🚢

File: .changeset/lucky-insects-report.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@khanacademy/wonder-blocks-core": minor
+---
+
+useIsMounted() hook
```
Comment (somewhatabstract) [inline .changeset/lucky-insects-report.md:5]: **nit:** I find our release notes more useful if they have a verb explaining what changed. Like "Added xxx" or "Fixed xxx"

File: packages/wonder-blocks-core/src/hooks/__tests__/use-is-mounted.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// @flow
+import {renderHook} from "@testing-library/react-hooks";
+
+import {useIsMounted} from "../use-is-mounted.js";
+
+describe("useIsMounted", () => {
+    it("should return false on first call", () => {
```
Comment (somewhatabstract) [inline packages/wonder-blocks-core/src/hooks/__tests__/use-is-mounted.test.js:7]: **suggestion:** The Arrange/Act/Assert comments to divide sections are missing here.

See best practices: https://khanacademy.atlassian.net/wiki/spaces/ENG/pages/98402353/JavaScript+Testing+Best+Practices#Arrange-Act-Assert

File: packages/wonder-blocks-core/src/hooks/use-is-mounted.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// @flow
+import * as React from "react";
+
+/**
+ * Returns a function that can be called to check if the current component is still
```
Comment (somewhatabstract) [inline packages/wonder-blocks-core/src/hooks/use-is-mounted.js:5]: **suggestion:** We don't tend to comment this way.

Consider:

```
/**
 * Hook to provide a function for determining component mounted state.
```

File: packages/wonder-blocks-core/src/hooks/use-is-mounted.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * mounted or not.
+ *
+ * NOTE: Based on https://github.com/juliencrn/usehooks-ts/blob/d5f3de88cc319c790f2a4e90ba6a8904298957a5/src/useIsMounted/useIsMounted.ts
+ *
+ * @returns {() => boolean}
```
Comment (somewhatabstract) [inline packages/wonder-blocks-core/src/hooks/use-is-mounted.js:10]: **nit:** Usually there's a comment string explaining what this is.

```
@returns {() => boolean} A function that returns the component mounted state.
```

File: packages/wonder-blocks-core/src/hooks/use-is-mounted.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+ */
+export const useIsMounted = (): (() => boolean) => {
+    const isMounted = React.useRef<boolean>(false);
+    React.useEffect(() => {
+        isMounted.current = true;
+        return () => {
+            isMounted.current = false;
+        };
+    }, []);
```
Comment (somewhatabstract) [inline packages/wonder-blocks-core/src/hooks/use-is-mounted.js:19]: **question:** Why not use `useOnMountEffect`?

**Review by somewhatabstract — APPROVED** [review]
Generally looks good, though there are some nits to be addressed before landing.

File: packages/wonder-blocks-core/src/hooks/use-is-mounted.js (hunk 5 above)
Comment (kevinbarabash) [reply packages/wonder-blocks-core/src/hooks/use-is-mounted.js:19]: Good idea.  That PR hasn't landed yet.  I'll land that one first and then update this file to use.

**Pull request merged**